### PR TITLE
C#8 using to reduce nesting

### PR DESCRIPTION
**Pull request by gerhardol**
Part of #8728

## Proposed changes
 
using declarations, to reduce nestings
A few variables have slightly wider declaration now
A few uses of using cannot be changed, like in switch statements.

## Test methodology 

manual, code review
Directory diff with kdiff3 works OK

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: GitUI/Theming/Renderers/HeaderRenderer.cs [hunk 1]
```diff
@@ -11,17 +11,16 @@ internal class HeaderRenderer : ThemeRenderer
...
         {
-            using (var ctx = CreateRenderContext(hdc, pcliprect))
+            using var ctx = CreateRenderContext(hdc, pcliprect);
+            switch ((Parts)partId)
             {
-                switch ((Parts)partId)
-                {
-                    case Parts.None:
+                case Parts.None:
                     {
```
Comment (mstv) [inline GitUI/Theming/Renderers/HeaderRenderer.cs:18]: Why not unindent this code block, too - and a few more below?

File: GitUI/Theming/Renderers/ButtonRenderer.cs [hunk 2]
```diff
@@ -164,10 +164,10 @@ private int RenderCheckBox(Context ctx, State.CheckBox stateid, Rectangle prect)
...
-                    using (ctx.HighQuality())
-                    {
-                        ctx.Graphics.DrawLines(checkPen, points);
-                    }
+                        using (ctx.HighQuality())
+                        {
+                            ctx.Graphics.DrawLines(checkPen, points);
+                        }
 
                     break;
```
Comment (mstv) [inline GitUI/Theming/Renderers/ButtonRenderer.cs:172]: For consistency with below changes:
```
                    using (ctx.HighQuality())
                    {
                        using var checkPen = new Pen(foreColor, DpiUtil.Scale(1.5f));
                        ctx.Graphics.DrawLines(checkPen, points);
                    }
                    break;
```

**Review by mstv — APPROVED** [review]

File: GitUI/Theming/Renderers/HeaderRenderer.cs (hunk 1 above)
Comment (gerhardol) [reply GitUI/Theming/Renderers/HeaderRenderer.cs]: I only left this to make sure someone reviewed - or it was something where autoindent is not working properly and I missed to to fix this manually after reverting.
Good catch, 
Pushed fixup.

File: GitUI/Theming/Renderers/ButtonRenderer.cs (hunk 2 above)
Comment (gerhardol) [reply GitUI/Theming/Renderers/ButtonRenderer.cs]: I missed this snippet, thanks.

**Comment by mstv** [comment]
> A few variables have slightly wider declaration now

I agree that these should be uncritical.

**Review by RussKie — APPROVED** [review]

**Comment by RussKie** [comment]
Thank you

**Comment by gerhardol** [comment]
@RussKie Merged with a fixup... Is it possible to block that?

**Comment by pmiossec** [comment]
https://github.com/marketplace/actions/block-fixup-commit-merge

**Comment by RussKie** [comment]
Yes, sorry, realised after hit the merge button.

**Comment by RussKie** [comment]
Thank you @pmiossec. I installed the check:
image

**Pull request merged**